osbs/build/build_request.py

```diff
@@ -1138,6 +1147,23 @@ def render_customizations(self):
     def render_version(self):
         self.dj.dock_json_set_param('client_version', client_version)
 
+    def render_node_selectors(self):
+        # for worker builds set nodeselectors
+        if self.spec.platforms.value is None:
+
+            # auto or explicit build selector
+            if self.is_auto:
+                self.template['spec']['nodeSelector'] = self.auto_build_node_selector
+            else:
+                self.template['spec']['nodeSelector'] = self.explicit_build_node_selector
+                # scratch build nodeselector
+                if self.scratch:
+                    self.template['spec']['nodeSelector'].update(self.scratch_build_node_selector)
```

Comment:
why do we want scratch and explicit node selectors for the same build?
The intent was to have either one of the three types set, not a combination of some.